mapslang/Makefile

```diff
@@ -1,27 +1,96 @@
-  SLANG_HOST=mingw32
+  ifneq ($(BUILD_FOR_LINUX), 1)
+    ## build slang-pre2.3.3-24/stc/mkfiles/mkmake.c for linux, only the .exe is provided in the slang package
+    ## This is used to enable mingw32 DLL builds in linux
+    SLANG_BUILDCOMMAND = cp ./src/slconfig.h ./src/config.h && \
+                         gcc -I./src ./src/mkfiles/mkmake.c ./src/slprepr.c -o ./src/mkfiles/mkmake &&
+  else
+    SLANG_BUILDCOMMAND = ./configure && make
+  endif
 endif
-
-SLANG_URL = https://www.jedsoft.org/snapshots/slang-pre2.3.3-24.tar.gz
-SLANG_NAME=slang-pre2.3.3-24
-
-CC                = $(CROSS_COMPILE)gcc
-CPP                = $(CROSS_COMPILE)g++
-CFLAGS                          = -c -O -I. -I../lib/$(SLANG_NAME)/src -DSLANG_DLL=1
-CFLAGS-DEBUG                    = -c -g -I. -I../lib/$(SLANG_NAME)/src -DSLANG_DLL=1
-LINK                = $(CROSS_COMPILE)gcc
-WINDRES                         = $(CROSS_COMPILE)windres
-LFLAGS                          = 
-LFLAGS-DEBUG                    = 
-LIBS                = -L. -L../libadikted -L../lib/$(SLANG_NAME)/src/gw32objs -lslang -ladikted
+# Executable file extensions on target environment
+BUILD_FOR_LINUX ?= 0
+ifeq ($(BUILD_FOR_LINUX), 1)
+  EXEEXT =
+  OTHER_EXEEXT = .exe
+  LIBEXT = .so
+  OTHER_LIBEXT = .dll
+  DLL_FLAG =
+  SLANG_OBJ_FOLDER = elfobjs
+else
+  EXEEXT = .exe
+  OTHER_EXEEXT =
+  LIBEXT = .dll
+  OTHER_LIBEXT = .so
+  DLL_FLAG = -DSLANG_DLL=1
+  SLANG_OBJ_FOLDER = gw32objs
+  # S-Lang MinGW32 Makefile creation:
+  # -static-libgcc is also needed by libslang. So the 3 makefile.all files are patched.
+  # (Along with tweaking 'del'/'rm' to 'rm -f' for the clean process, and allowing TOOL_PREFIX, 
+  # CFLAGS, and LFLAGS to be passed to the compiler/linker from this Makefile.)
+  SLANG_PLATFORM = WIN32 MINGW32 DLL
+  SLANG_CFLAGS = -std=gnu89
+  SLANG_LFLAGS = -static-libgcc
+  SLANG_RC_FOLDER = ../../../mapslang/libslang-patches
+  SLANG_BUILDCOMMAND += \
+  patch -r - -N src/mkfiles/makefile.all ../../mapslang/libslang-patches/slang-src-mingw32-makefile.patch && \
+  patch -r - -N slsh/mkfiles/makefile.all ../../mapslang/libslang-patches/slang-slsh-mingw32-makefile.patch && \
+  patch -r - -N modules/mkfiles/makefile.all ../../mapslang/libslang-patches/slang-modules-mingw32-makefile.patch && \
+  TOOL_PREFIX='$(CROSS_COMPILE)' CFLAGS='$(SLANG_CFLAGS)' LFLAGS='$(SLANG_LFLAGS)' SLANG_RC_FOLDER='$(SLANG_RC_FOLDER)' mkfiles/build.sh $(SLANG_PLATFORM)
+endif
+# Allow debug executables to be created
+DEBUG ?= 0
+ifeq ($(DEBUG), 1)
+  DEBUGME = yes
+else
+  DEBUGME = no
+endif
+# Names of utility commands
+SLANG_URL          = https://www.jedsoft.org/snapshots/slang-pre2.3.3-24.tar.gz
+SLANG_NAME         = slang-pre2.3.3-24
+SLANG_ARCHIVE      = ../lib/slang.tgz
+SLANG_BIN_FOLDER   = /lib/$(SLANG_NAME)/src/$(SLANG_OBJ_FOLDER)
+WARNFLAGS          = -Wall -Wextra -Wno-pedantic -Wno-conversion -Wno-traditional-conversion -Wno-sign-compare
+PLATFORM_FLAGS     = -march=i486
```

Comment:
Come to think of it, this doesn't make much sense. If you were to actually run this on an CPU matching this arch (a CPU from before 1997) it's likely going to crash because its dependency `libadikted` is compiled with MMX instructions support.

To be fair this is such an extremely fringe edge case and was grandfathered by the old Makefiles, I'm not consider it blocking for a merge. Fix would be to at minimum match `-march` of `libadikted` and perhaps add `-mmmx`.